src/main/java/com/solvd/app/utils/DOMParser.java

```diff
@@ -0,0 +1,92 @@
+package com.solvd.app.utils;
+
+import com.solvd.app.models.*;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.w3c.dom.*;
+
+import javax.xml.parsers.*;
+import java.util.function.Function;
+
+public class DOMParser {
+
+    private static final Logger LOGGER = LogManager.getLogger(DOMParser.class);
+
+    public static void parseXMLFile(String xmlFilePath) {
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+
+        try {
+            DocumentBuilder builder = factory.newDocumentBuilder();
+            Document doc = builder.parse(xmlFilePath);
+
+            parseElements(doc, "pharmacy", DOMParser::parsePharmacy);
+            parseElements(doc, "person", DOMParser::parsePerson);
+            parseElements(doc, "position", DOMParser::parsePosition);
+            parseElements(doc, "staff", DOMParser::parseStaff);
+            parseElements(doc, "customer", DOMParser::parseCustomer);
+        } catch (Exception e) {
+            LOGGER.error("Error while parsing XML file", e);
+        }
+    }
+
+    private static void parseElements(Document doc, String tagName, Function<Element, Object> parserFunction) {
+        NodeList nodes = doc.getElementsByTagName(tagName);
+        for (int i = 0; i < nodes.getLength(); i++) {
+            Element element = (Element) nodes.item(i);
+            Object parsedObject = parserFunction.apply(element);
+            LOGGER.info(parsedObject);
```

Comment:
might be better to separate logging from parsing logic since there may be more important things you can do with the parsedObject

Reply:
To avoid increasing complexity, I will keep the current implementation as it is for now. Thank you for pointing it out.